client/state/current-user/email-verification/reducer.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * External dependencies
+ */
+import { createReducer } from 'state/utils';
+// import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import {
+	EMAIL_VERIFY_REQUEST,
+	EMAIL_VERIFY_REQUEST_SUCCESS,
+	EMAIL_VERIFY_REQUEST_FAILURE,
+	EMAIL_VERIFY_STATE_RESET,
+} from 'state/action-types';
+
+export default createReducer( {}, {
+	[ EMAIL_VERIFY_REQUEST ]: () => ( {
+		pendingRequest: true,
+		emailSent: false,
+		error: false,
```

Comment:
separating out is definitely more ideal in many ways - keeps the state simple and composable vs mangled. it's worth working with the data layer and seeing how much easier the reducers might get